src/Tutor.Infrastructure/Interceptors/Logging.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using Tutor.Core.BuildingBlocks;
+using Castle.DynamicProxy;
+using FluentResults;
+using System;
+
+namespace Tutor.Infrastructure.Interceptors
+{
+    public class Logging : IInterceptor
+    {
+        private readonly ILogger<Logging> _logger;
+
+        public Logging(ILogger<Logging> logger)
+        {
+            _logger = logger;
+        }
+
+        public void Intercept(IInvocation invocation)
+        {
+            _logger.LogDebug("Calling method {@Class}.{@Method}.", 
+                invocation.TargetType, invocation.Method.Name);
+            invocation.Proceed();
+            try
+            {
+                dynamic result = invocation.ReturnValue;
+                if (result.IsSuccess)
+                {
+                    _logger.LogInformation("Successful execution of method: {@Class}.{@Method}.", 
```

Comment:
The log before method execution is Debug level, and because of the memory restrictions that were discussed, default level for production was to be Warning. Therefore, that log would not be written. Its purpose was to allow view of method execution time while developing a feature. As this is a rare use case, I will remove it.
Other suggestions are implemented as well.